src/packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -457,7 +466,12 @@ export function getPrismaClient(config: GetPrismaClientOptions): any {
       return this.$connect()
     }
     async $connect() {
-      return this._engine.start()
+      try {
+        return this._engine.start()
+      } catch (e) {
+        e.clientVersion = this._clientVersion
+        throw e
+      }
     }
```

Comment:
Added some more try-catches in the top-level methods, such as connect, disconnect, executeRaw, queryRaw & transaction. While there's already some errors caught at the `PrismaClientFetcher.request` level, a lot of stuff that could go wrong is happening before that. If an error is caught/rethrown at the `PrismaClientFetcher.request` level, top-level methods should also catch it and re-throw the error, without touching it this time.

Reply:
Sounds good :)